pkg/util/progressreader/progressreader.go

```diff
@@ -0,0 +1,73 @@
+package progressreader
+
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+// ProgressReader wraps an existing io.ReadCloser.
+type ProgressReader struct {
+	io.ReadCloser
+	total         int64 // Total # of bytes transferred
+	transferEnd   time.Time
+	transferStart time.Time
+	transferStep  time.Time
+}
+
+// Override Read method in order to print some progress
+// during reading.
+func (r *ProgressReader) Read(p []byte) (int, error) {
+	if r.transferStart.IsZero() {
+		// initialize start timer
+		r.transferStart = time.Now()
+	}
+
+	curr, err := r.ReadCloser.Read(p)
+	r.total += int64(curr)
+
+	if err == nil {
+		// initialize the transfer current step, or check if at least 1s is passed
+		// in order to not spam progress updates too much
+		if r.transferStep.IsZero() || time.Now().Second() >= (r.transferStep.Second()+1) {
+			r.transferStep = time.Now()
+
+			fmt.Printf("\n\033[1A\033[K... uploaded %s\r", toHumanReadable(r.total))
+		}
+	}
+
+	if err == io.EOF {
+		r.transferEnd = time.Now()
+	}
+
+	return curr, err
+}
+
+// Rate returns the rate of progress in b/s
+func (r *ProgressReader) Rate() string {
```

Comment:
Is this used anywhere?

Reply:
Not for now, Initially I've put this as it would be useful to track transfer speeds?
Could be added to the info?

Like:
![image](https://user-images.githubusercontent.com/598882/215548277-c35505c3-8ba0-4a5c-b661-c58d3f69a0c6.png)